### PR TITLE
Acceptance tests for Application Gateway

### PR DESCRIPTION
**Pull request by Szymongib**
**Description**

Changes proposed in this pull request:

- Initial implementation of Application Gateway Tests

**Related issue(s)**
#1589

### CODE REVIEW

File: tests/application-gateway-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+The project requires Go 1.8 or higher.
+
+## Usage
+
+Environment parameters used by the tests:
+
+| Name | Required | Default | Description | Possible values |
+|------|----------|---------|-------------|-----------------|
+| **APPLICATION** | Yes | - | The name of the Application to test | `my-application` | 
+| **NAMESPACE** | Yes | - | The namespace on which the test application will operate | `kyma-integration` |
```
Comment (tomekpapiernik) [inline tests/application-gateway-tests/README.md]:
```suggestion
| **NAMESPACE** | Yes | - | The Namespace in which the test Application will operate | `kyma-integration` |
```

File: tests/application-gateway-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+| **MOCK_SERVICE_PORT** | Yes | - |  Port number used by the mock service created by the test | `8080` |
+| **TEST_EXECUTOR_IMAGE** | No | Same version as the Helm test image  | Image name of the test executor created by the Helm test  | `user/my-image:1.0.0` |
+
+
+## Development
+
+Project contains of two applications Helm Test and Test Executor.
```
Comment (tomekpapiernik) [inline tests/application-gateway-tests/README.md]:
```suggestion
The project consists of two applications: Helm Test and Test Executor.
```

File: tests/application-gateway-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+Test Executor pod is managed by Helm Test.
```
Comment (tomekpapiernik) [inline tests/application-gateway-tests/README.md]:
```suggestion
The Test Executor Pod is managed by Helm Test.
```

File: tests/application-gateway-tests/test/executor/proxy/suite.go [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+	assert.NoError(t, err)
+
+	ts.deleteMockService(t)
+}
+
+func (ts *TestSuite) ApplicationName() string {
+	return ts.config.Application
+}
+
+// WaitForAccessToAPIServer waits for access to API Server which might be delayed by initialization of Istio sidecar
```
Comment (janmedrek) [inline tests/application-gateway-tests/test/executor/proxy/suite.go]: this comment seems a bit obsolete, we can remove it

Comment (Szymongib) [reply tests/application-gateway-tests/test/executor/proxy/suite.go]: I do not think it is obsolete. I think it is quite useful as it seems to me that it is not totally obvious why we wait for access to API Server.

Comment (janmedrek) [reply tests/application-gateway-tests/test/executor/proxy/suite.go]: Let's leave it then :)

**Pull request merged**